src/v/cluster/metadata_dissemination_handler.cc

```diff
@@ -41,17 +44,24 @@ metadata_dissemination_handler::update_leadership_v2(
 
 ss::future<update_leadership_reply>
 metadata_dissemination_handler::do_update_leadership(
-  std::vector<ntp_leader_revision> leaders) {
+  ss::chunked_fifo<ntp_leader_revision> leaders) {
     vlog(clusterlog.trace, "Received a metadata update");
-    return _leaders
-      .invoke_on_all(
-        [leaders = std::move(leaders)](partition_leaders_table& pl) mutable {
-            for (auto& leader : leaders) {
-                pl.update_partition_leader(
-                  leader.ntp, leader.revision, leader.term, leader.leader_id);
-            }
-        })
-      .then([] { return ss::make_ready_future<update_leadership_reply>(); });
+    for (ss::shard_id i = 0; i < ss::smp::count; ++i) {
```

Comment:
whats the rationale behind unwrapping this loop? I think invoke_on_all  does a parallel invoke on shards.

Reply:
yes, but the `invoke_on_all` copy lambda to each shard, we need to copy the `ss::chunked_fifo` explicitly as it is not copyable.

Reply:
Ahhhh, right.. guess we could when_all() on sumbit_to() futures to preserve parallelism..